codegen/src/main/kotlin/norm/codegen/CodeGenerator.kt

```diff
@@ -16,7 +16,8 @@ class CodeGenerator(private val typeMapper: DbToKtDefaultTypeMapper = DbToKtDefa
         val paramSetterClassName = "${baseName}ParamSetter"
 
         val fileBuilder = FileSpec.builder(packageName, baseName)
-
+        val tableCount = cols.distinctBy { it.tableName }.count()
+        val primaryTableName = cols.distinctBy { it.tableName }.first().tableName
```

Comment:
Is it guaranteed that first table is always the primary table?

Reply:
@bhimsenp it depends on the sequence of list of columns ,need to check if we deliberately specify columns from second table before primary, will try adding a test for it :) 👍

Reply:
need other way to find primary table 🤔

Reply:
We also need to think about differentiating left & inner joins. Because for inner join, nullability will remain as it is.

Reply:
Any updates on this @bhimsenp @swanandvk?

Reply:
The current solution would not work.
@swanandvk , can we close this if there is no further progress?

Reply:
yup 👍